src/statmodels.jl

```diff
@@ -76,6 +76,26 @@ This is usually the model containing only the intercept.
 nullloglikelihood(model::StatisticalModel) =
     error("nullloglikelihood is not defined for $(typeof(model)).")
 
+"""
+    loglikelihood(model::StatisticalModel, ::Colon)
+
+Return a vector of each observation's contribution to the log-likelihood of the model.
+In other words, this is the vector of the pointwise log-likelihood contributions.
+
+
+In general, `sum(loglikehood(model, :)) == loglikelihood(model)`.
+"""
+loglikelihood(model::StatisticalModel, ::Colon) =
+    error("loglikelihood(model::StatisticalModel, ::Colon) is not defined for $(typeof(model)).")
+
+"""
+    loglikelihood(model::StatisticalModel, observation)
+
+Return the log-likelihood of `observation` contribution under `model`.
```

Comment:
That sentence feels off to me. Something like this?
```suggestion
Return the contribution of `observation` to the log-likelihood of `model`.
```